python/lsst/afw/geom/testUtils.py

```diff
@@ -91,3 +100,759 @@ def __iter__(self):
+        # and I don't want to modify the inputs, so replace the input frames with copies
+        baseFrame = baseFrame.copy()
+        baseFrame.setIdent("baseFrame")
+        currFrame = currFrame.copy()
+        currFrame.setIdent("currFrame")
+
+        frameSet = astshim.FrameSet(baseFrame)
+        frame2 = astshim.Frame(nIn, "Ident=frame2")
+        frameSet.addFrame(astshim.FrameSet.CURRENT, astshim.UnitMap(nIn), frame2)
+        frame3 = astshim.Frame(nOut, "Ident=frame3")
+        frameSet.addFrame(astshim.FrameSet.CURRENT, polyMap, frame3)
+        frameSet.addFrame(astshim.FrameSet.CURRENT, astshim.UnitMap(nOut), currFrame)
+        frameSet.setIdent("currFrame")
+        return frameSet
+
+    @staticmethod
+    def permFrameSetIter(frameSet):
+        """Iterator over 0 or more frameSets with SkyFrames axes permuted
+
+        Only base and current SkyFrames are permuted. If neither the base nor the
+        current frame is a SkyFrame then no frames are returned.
+
+        Each returned value is a tuple:
+        - permFrameSet: a copy of frameSet with the base and/or current frame permuted
+        - isBaseSkyFrame: a boolean
+        - isCurrSkyFrame: a boolean
+        - isBasePermuted: a boolean
+        - isCurrPermuted: a boolean
+        """
+        baseInd = frameSet.getBase()
+        currInd = frameSet.getCurrent()
+        isBaseSkyFrame = frameSet.getFrame(baseInd).getClass() == "SkyFrame"
+        isCurrSkyFrame = frameSet.getFrame(currInd).getClass() == "SkyFrame"
+        if not (isBaseSkyFrame or isCurrSkyFrame):
+            return
+
+        baseList = [False, True] if isBaseSkyFrame else [False]
+        currList = [False, True] if isCurrSkyFrame else [False]
+        for isBasePermuted in baseList:
+            for isCurrPermuted in currList:
+                frameSetCopy = frameSet.copy()
+                if isBasePermuted:
+                    assert isBaseSkyFrame
+                    frameSetCopy.setCurrent(baseInd)
+                    frameSetCopy.permAxes([2, 1])
+                    frameSetCopy.setCurrent(currInd)
+                if isCurrPermuted:
+                    assert isCurrSkyFrame
+                    frameSetCopy.permAxes([2, 1])
+                yield (frameSetCopy, isBaseSkyFrame, isCurrSkyFrame, isBasePermuted, isCurrPermuted)
+
+    @staticmethod
+    def makeJacobian(nIn, nOut, inArray):
+        """Make a Jacobian matrix for the equation described by makeTwoWayPolyMap.
+
+        Parameters
+        ----------
+        nIn, nOut : integers
+            the dimensions of the input and output data; see makeTwoWayPolyMap
+        inArray : ndarray
+            an array of size `nIn` representing the point at which the Jacobian
+            is measured
+
+        Returns
+        ----------
+        J : numpy.ndarray
+            an nOut x nIn array of first derivatives
+        """
+        baseCoeff = 2.0 * 0.001
```

Comment:
Who did write the code?  It seems scary to have code we don't understand at this point in the package's history.